task_geo/data_sources/nasa/nasa.py

```diff
@@ -0,0 +1,214 @@
+    region : string
+    sub_region : string
+    lat : string
+    lon : string
+    str_start_date : string
+    str_end_date : string
+    parms_str : string
+
+    Returns
+    -------
+    df : pandas.DataFrame
+
+    """
+    url = f"{base_url}?request=execute&{identifier}&{parms_str}&" \
+          f"startDate={str_start_date}&endDate={str_end_date}&" \
+          f"lat={lat}&lon={lon}&{temporal_average}&{output_format}&" \
+          f"{user_community}&{user}"
+    response = requests.get(url)
+    data_json = response.json()
+    df = pd.DataFrame(data_json['features'][0]['properties']['parameter'])
+    df['country'] = country
+    df['region'] = region
+    df['sub_region'] = sub_region
+    df['lon'] = lon
+    df['lat'] = lat
+    return df
+
+
+def nasa_connector(df_locations, start_date, end_date=None,
+                   parms=["temperature", "humidity", "pressure"],
+                   level="region"):
+    """
+    Retrieve meteorologic data from NASA.
+
+    Given a dataset with columns country, region, sub_region, lon, and lat, for
+    each geographic coordinate (lon, lat) corresponding to a place (specified
+    if country, region, or sub_region) extract the time series of the desired
+    data at the location.
+
+    Arguments:
+    ---------
+        df_locations(pandas.DataFrame): Dataset with columns country, region,
+                                        sub_region, lon, and lat
+        start_date(datetime64): Start date for the time series
+        end_date(pandas.Timestamp): End date fo rthe time series (optional)
+        parms(list of strings): Desired data, accepted are 'temperature',
+                                'humidity', and 'pressure' (optional)
+        level(string):  country, region, or sub_region, tells us at what
+                        granularity we want the data (eg for US sub_region, ie
+                        county, might make more sense than region, ie states,
+                        but in France region could be better, while for
+                        Luxembourg we probably want country) (optional)
+
+    Return:
+    ------
+        pandas.DataFrame:   Columns are country, region, sub_region (non-null),
+                            lon, lat, date, and the desired data.
+    """
+    df_locations = df_locations[
+        ~pd.isna(df_locations[[level, 'lon', 'lat']]).all(axis=1)
+    ]
+    location_data = ['country', 'region', 'sub_region', 'lon', 'lat']
+    locations = df_locations[location_data].drop_duplicates()
+
+    str_start_date = str(start_date.date()).replace('-', '')
+
+    if end_date is None:
+        str_end_date = str(pd.Timestamp.today().date()).replace('-', '')
+    else:
+        str_end_date = str(end_date.date()).replace('-', '')
+
+    all_parms = list(itertools.chain.from_iterable([PARAMETERS[p]
+                                                    for p in parms]))
+    parms_str = f"parameters={','.join(all_parms)}"
+
+    return pd.concat([
```

Comment:
Couple of things here:
1. Country, region and subregion are only used to mark the result dataset, not to call NASA,  you can not send them to the function and then make a merge after the result is retrieved.

Reply:
"you can not send them to the function" -> "You shouldn't pass them to the function".

Sorry, as you know, my python is fire, but my english is potato.

Reply:
Done.